Install agent from sources

jcvenegas opened the pull request:
Instead of wait Clear Linux to update agent install it from sources.

jcvenegas committed:
rootfs: Query runtime repository for Clear Linux version.

Query to know what is the base recommend to build the rootfs.

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

jcvenegas committed:
rootfs: Create file with information about image

show packages intalled

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

jcvenegas committed:
rootfs: build cc-agent from source

Instead of install cc-agent package from Clear Linux
install the latest master.

Fixes: #38

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

jcvenegas committed:
packages: remove bootchart

bootchart is not used by our automation to measure boot time.

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

jcvenegas committed:
packages: add clr-power-tweaks

Add clr-power-tweaks to add networking multiqueue.

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

devimc commented on Makefile:
```diff
@@ -102,7 +108,10 @@ docker-build:
-	rm -rf "$(WORKDIR)/rootfs" "$(WORKDIR)/linux" "$(WORKDIR)/img"
+	sudo rm -rf "$(WORKDIR)/rootfs"
+	rm -rf "$(WORKDIR)/linux"
+	rm -rf "$(WORKDIR)/img"
```
Q. should this be ```sudo``` too?

jcvenegas replied:
I am adding just to linux because the rootfs files may be owned by root (it is a a rootfs). I dont think the rest are need to have sudo.

devimc commented on scripts/Dockerfile:
```diff
@@ -17,13 +17,7 @@
 From fedora:27
 
-RUN dnf install -y qemu-img parted gdisk make gcc bc git e2fsprogs
+RUN dnf install -y qemu-img parted gdisk make gcc bc git e2fsprogs go libudev-devel
```
do you know what version of go is installed?

jcvenegas replied:
I'dont, I can install it from golang compiled binaries to have a version we know it works.

jodh-intel replied:
Good point @devimc. I agree @jcvenegas - that is going to be the safest approach.

devimc commented on scripts/config.txt:
```diff
@@ -0,0 +1,7 @@
+#Clear Linux version to use as base rootfs
```
I think this should be part of runtime versions.txt, right?

jcvenegas replied:
Yes, I feel that is a better idea. instead of having to maintain multiples versions files I can point or could to know what is the version to use,  reworking that.

jcvenegas commented on scripts/osbuilder.sh:
```diff
@@ -24,8 +24,17 @@ fi
+VESIONS_FILE="https://raw.githubusercontent.com/clearcontainers/runtime/master/versions.txt"
+
+if [ -z "${OS_VERSION}" ]; then
+	OS_VERSION="$(curl -Ls "${VESIONS_FILE}" | grep '^clear_vm_image_version=' | cut -d= -f 2)"
```
Now query to runtime versions.txt file.

jcvenegas replied:
@devimc  ^^

jodh-intel commented on Makefile:
```diff
@@ -26,10 +26,14 @@ BUILD_PROXY += --build-arg https_proxy=$(https_proxy)
 endif
 
+RUN_OS_VERSION= --env OS_VERSION="$(OS_VERSION)"
+RUN_AGENT_VERSION= --env AGENT_VERSION=$(AGENT_VERSION)
```
Missing quotes for the variable value (for consistency with the other variables around this one)?

jodh-intel commented on Makefile:
```diff
@@ -26,10 +26,14 @@ BUILD_PROXY += --build-arg https_proxy=$(https_proxy)
 RUN_REPO_URL = --env REPO_URL="$(REPO_URL)"
 RUN_DEBUG = --env DEBUG="$(DEBUG)"
+CONTAINER_GOPATH = /go_path
+RUN_GOPATH += --env GOPATH=$(CONTAINER_GOPATH)
```
Again, prolly best to add quotes to the value here.

jcvenegas committed:
dockerfile: Install recommended golang from runtime

Instead of install golang from fedora install the version recommended by
the runtime.

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

jcvenegas commented:
@jodh-intel @devimc now golang installed from based on versions from runtime.

jodh-intel commented:
Thanks @jcvenegas.

lgtm

devimc commented:
lgtm
